docs/guides/custom-images.rst

```diff
@@ -0,0 +1,78 @@
+============================================
+
+When running workflows remotely on Compute Engine, your tasks run in a container that uses one of the
+``orquestra-sdk-base`` images depending on whether you have requested a GPU or not (you can see their source code
+`here <https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/main/docker/Dockerfile>`_ and
+`here <https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/main/docker/cuda.Dockerfile>`_). These images
+are automatically selected based on your workflow's SDK version. If your task needs extra dependencies that you
+cannot install using imports (such as native libraries or executable binaries), you can use one of the above images
+as your base image to publish a new image that you will use as your container when running your task.
+
+Building an Image
+-----------------
+
+The base images are hosted at Orquestra Nexus. There are two versions of the base image per each SDK version, one with
+CUDA dependencies installed and one without. Their names follow the pattern
+``hub.nexus.orquestra.io/zapatacomputing/orquestra-sdk-base:<SDK version>[-cuda]``. For example,
+``hub.nexus.orquestra.io/zapatacomputing/orquestra-sdk-base:0.48.0`` or
+``hub.nexus.orquestra.io/zapatacomputing/orquestra-sdk-base:0.49.0-cuda``. You should pick the one that matches
+your SDK version and your GPU needs (``-cuda`` needs to be picked for GPU workflows).
+
+.. note::
+
+    It is your responsibility as the author of the custom image to keep your image up to date with new SDK versions.
+
+The base images run as a user named ``orquestra`` with uid ``1000``. Since these are Ubuntu based images, you
+can install any package that you need from Ubuntu repositories (by doing ``RUN apt install <package name>``).
+However, before doing so, you need to temporarily switch to the root user (via ``USER root`` directive) and
+restore back to user ``orquestra`` once you're done (``USER orquestra``).
+
+.. note::
+
+    You can use `standard OCI annotations <https://github.com/opencontainers/image-spec/blob/main/annotations.md>`_ to add metadata to your images.
+
+
+In order for the Compute Engine to run your workflow code, you should not have a ``ENTRYPOINT`` or ``CMD`` statement in
+your images. In case you do, these will be overriden at run time and you might observe unexpected behavior.
+
+Publishing the Image
+--------------------
+
+Once you have your ``Dockerfile`` ready, you can publish your image to our Nexus repository by using standard Docker tools.
+Please refer to `this page <https://zapatacomputing.atlassian.net/wiki/spaces/~61209e4528ae75006af8a1b8/pages/619577422/Nexus+Starts+Here>`_
+to see how.
+
+..
+    TODO: Either move the page to a more general space or copy the relevant bits here
+
+
+Using the Image on Compute Engine
+---------------------------------
+
+You can run workflows that use custom images only on Compute Engine and not on Quantum Engine or locally. You
```

Comment:
Would be nice to clarify - possibly via example or a link to a place that does - what does it mean to run on Compute Engine. To end users, I understand that means they must use `--ce` option (soon to be the default) when logging into an orquestra remote cluster.

Reply:
I considered that but I found no place in the docs that refer to the `--ce` option. Since it's a central piece and might change the structure of the pages, I think it's best if someone from the SDK team adds such a section and then we can link it here.

Reply:
Done in ORQSDK-857